File: backend-django/projects/management/commands/cleanup_project_pages.py
"""
Management command to cleanup project pages - keep only specified pages
"""
from django.core.management.base import BaseCommand
from projects.models import Project, ProjectPage


class Command(BaseCommand):
    help = 'Retire toutes les pages d\'un projet sauf celles spécifiées'

    def add_arguments(self, parser):
        parser.add_argument(
            'project_id',
            type=int,
            help='ID du projet à nettoyer'
        )
        parser.add_argument(
            '--keep',
            nargs='+',
            default=['home', 'test'],
            help='Slugs des pages à conserver (défaut: home test)'
        )

    def handle(self, *args, **options):
        project_id = options['project_id']
        keep_slugs = set(options['keep'])
        
        try:
            project = Project.objects.get(id=project_id)
            self.stdout.write(f'📁 Projet: {project.name} (ID: {project.id})')
            
            # Récupérer toutes les pages du projet
            all_pages = ProjectPage.objects.filter(project=project)
            total_pages = all_pages.count()
            
            self.stdout.write(f'📄 Total de pages dans le projet: {total_pages}')
            
            # Filtrer les pages à retirer
            pages_to_remove = all_pages.exclude(page_slug__in=keep_slugs)
            pages_to_keep = all_pages.filter(page_slug__in=keep_slugs)
            
            self.stdout.write(f'✅ Pages à conserver: {pages_to_keep.count()}')
            for page in pages_to_keep:
                self.stdout.write(f'   - {page.page_slug} (ID: {page.id})')
            
            self.stdout.write(f'🗑️  Pages à retirer: {pages_to_remove.count()}')
            for page in pages_to_remove:
                self.stdout.write(f'   - {page.page_slug} (ID: {page.id})')
            
            if pages_to_remove.count() > 0:
                # Confirmer
                self.stdout.write(self.style.WARNING('\n⚠️  Voulez-vous vraiment retirer ces pages ?'))
                confirm = input('Tapez "oui" pour confirmer: ')
                
                if confirm.lower() == 'oui':
                    # Retirer les pages
                    removed_count = 0
                    for page in pages_to_remove:
                        page.delete()
                        removed_count += 1
                        self.stdout.write(self.style.SUCCESS(f'   ✅ Page "{page.page_slug}" retirée'))
                    
                    self.stdout.write(self.style.SUCCESS(f'\n✅ {removed_count} page(s) retirée(s) avec succès !'))
                    self.stdout.write(f'📄 Pages restantes dans le projet: {ProjectPage.objects.filter(project=project).count()}')
                else:
                    self.stdout.write(self.style.ERROR('❌ Opération annulée'))
            else:
                self.stdout.write(self.style.SUCCESS('\n✅ Aucune page à retirer. Le projet contient déjà uniquement les pages souhaitées.'))
                
        except Project.DoesNotExist:
            self.stdout.write(self.style.ERROR(f'❌ Projet avec ID {project_id} non trouvé'))
        except Exception as e:
            self.stdout.write(self.style.ERROR(f'❌ Erreur: {str(e)}'))

File: scripts/cleanup-project-pages.js
/**
 * Script pour nettoyer les pages d'un projet
 * Garde uniquement les pages spécifiées (home et test par défaut)
 * 
 * Usage: node scripts/cleanup-project-pages.js [projectId] [page1] [page2] ...
 * Exemple: node scripts/cleanup-project-pages.js 1 home test
 */

const axios = require('axios');

const API_BASE_URL = process.env.API_BASE_URL || 'http://localhost:8000/api';
const AUTH_TOKEN = process.env.AUTH_TOKEN || ''; // À définir

async function cleanupProjectPages(projectId, keepSlugs = ['home', 'test']) {
  try {
    console.log(`📁 Nettoyage du projet ID: ${projectId}`);
    console.log(`✅ Pages à conserver: ${keepSlugs.join(', ')}`);
    
    // 1. Récupérer le projet
    const projectResponse = await axios.get(`${API_BASE_URL}/projects/${projectId}/`, {
      headers: {
        'Authorization': `Bearer ${AUTH_TOKEN}`,
        'Content-Type': 'application/json'
      }
    });
    
    const project = projectResponse.data;
    console.log(`📁 Projet: ${project.name}`);
    
    if (!project.pages || project.pages.length === 0) {
      console.log('✅ Aucune page dans le projet');
      return;
    }
    
    console.log(`📄 Total de pages: ${project.pages.length}`);
    
    // 2. Identifier les pages à retirer
    const pagesToRemove = project.pages.filter(
      page => !keepSlugs.includes(page.page_slug)
    );
    
    const pagesToKeep = project.pages.filter(
      page => keepSlugs.includes(page.page_slug)
    );
    
    console.log(`\n✅ Pages à conserver (${pagesToKeep.length}):`);
    pagesToKeep.forEach(page => {
      console.log(`   - ${page.page_slug} (ID: ${page.id})`);
    });
    
    console.log(`\n🗑️  Pages à retirer (${pagesToRemove.length}):`);
    pagesToRemove.forEach(page => {
      console.log(`   - ${page.page_slug} (ID: ${page.id})`);
    });
    
    if (pagesToRemove.length === 0) {
      console.log('\n✅ Aucune page à retirer. Le projet contient déjà uniquement les pages souhaitées.');
      return;
    }
    
    // 3. Retirer les pages
    console.log('\n🗑️  Retrait des pages...');
    for (const page of pagesToRemove) {
      try {
        await axios.delete(`${API_BASE_URL}/projects/${projectId}/remove_page/`, {
          headers: {
            'Authorization': `Bearer ${AUTH_TOKEN}`,
            'Content-Type': 'application/json'
          },
          data: {
            page_id: page.id
          }
        });
        console.log(`   ✅ Page "${page.page_slug}" retirée`);
      } catch (error) {
        console.error(`   ❌ Erreur lors du retrait de "${page.page_slug}":`, error.response?.data || error.message);
      }
    }
    
    console.log(`\n✅ Nettoyage terminé ! ${pagesToRemove.length} page(s) retirée(s)`);
    
  } catch (error) {
    console.error('❌ Erreur:', error.response?.data || error.message);
    process.exit(1);
  }
}

// Récupérer les arguments
const args = process.argv.slice(2);
const projectId = parseInt(args[0]) || 1;
const keepSlugs = args.slice(1).length > 0 ? args.slice(1) : ['home', 'test'];

if (!AUTH_TOKEN) {
  console.error('❌ AUTH_TOKEN non défini. Définissez-le dans les variables d\'environnement.');
  console.log('   Exemple: export AUTH_TOKEN="votre_token"');
  process.exit(1);
}

cleanupProjectPages(projectId, keepSlugs);

